tests/neg/i5044b.scala

```diff
@@ -0,0 +1,8 @@
+class I0 {
+  class I1
+  def i3 = {
+    type T = i5.I1
+    val i4 = new T // error: `i5` is a forward reference extending over the definition of `i4`
+    val i5 = new I0
+  }
```

Comment:
Here is another test with a type lambda
```scala
class I0 {
  class I2[T1, T2]
  def test2 = {
    type A[T] = i5.I2[T, String]
    val i4 = new A[Int]
    val i5 = new I0
  }
}
```
This one currently crashes with Scala 2.12.6 😄

Reply:
I would also add:
```scala
  def test3 = {
    val x = new T // error: `T` is a forward reference extending over the definition of `x`
    val y = new I0
    type T = y.I1
  }
```